enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +35,294 @@ have the ability to apply user defined tags to many resources created by OpenShi
+   Final tag set to AWS resource = `key_infra1 = value_infra1`
+   Event action = An event is generated to notify user about the action status (success/failure) to create tags for the AWS resource.
+
+### Update tags scenarios
+Users can update the user-defined tags by editing `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io` type. New addition of tags are always appended.
+On update of `resourceTags`, AWS resource is not created or restarted.
+
+In the case where a user-defined tag is specified in the Infrastructure resource and
+1) There is already user-defined tag with the same key and value present for AWS resource, the user-define tag value for the AWS resource will not be updated.
+   For example,
+   Existing tag for AWS resource = `key_infra1 = value_update1`
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`
+   Action = There is no update for AWS resource.
+   Final tag set to AWS resource = `key_infra1 = value_update1`
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+2) There is already user-defined tag with the same key and different value present for AWS resource, the user-define tag value for the AWS resource will be updated.
+   For example,
+   Existing tag for AWS resource = `key_infra1 = value_old`
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`
+   Action = Existing tag for AWS resource is updated to reflect new value.
+   Final tag set to AWS resource = `key_infra1 = value_update1`
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+3) The new user-defined tag request has empty string `""`, the user-defined tag value for the AWS resource will be updated. Please refer to the `Delete Scenarios` for more details on empty value string handling.
+   For example,
+   Existing tag for AWS resource = `key_infra1 = value_old`
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`
+   Action = Existing tag for AWS resource is updated to reflect new value.
+   Final tag set to AWS resource = `key_infra1 = value_update1`
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about user-defined tag being marked for deletion.
+
+### Delete tags scenarios
+User-defined tags are deleted in two-pass method to avoid accidental deletion of tags by the user.
+- In the first pass, the user sets the user-defined tag value to empty string.
+- In the second pass, the user-defined tag is deleted only when following conditions are met,
+  1) the user-defined tag set to AWS resource has empty string value.
+  2) the user-defined tag set in `.spec.platformSpec.aws.resourceTags` has empty string.
```

Comment:
If a user sets the tag value to empty on the infrastructure resource, controllers will update to empty. 
In the second update or operator sync(), if the infrastructure resource still has the empty string, the resource will be deleted.

The protection here is about, if the user accidentally leaves the tag empty, user still has time to revert the change.

Reply:
> The protection here is about, if the user accidentally leaves the tag empty, user still has time to revert the change.

That's probably not sufficient protection, the likelihood that a resource is synced twice within a matter of seconds is very high in certain applications and as such, this two stage approach doesn't really buy the end user anything.

What's the negative effect of just removing the tag straight away? The user can always add it back no?

Reply:
Removing the tag for an empty string seems reasonable, but, is an empty value a valid option on AWS? If an empty value is allowed, then that won't work because an empty value could be a valid customer use case